adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,41 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source){
+    std::stack<char> valid;
+
+    for(const char & c : source){
+        
+        for(const auto & b : brackets ){
+            if(c==b.second && valid.empty()){
```

Comment:
please name this condition, it is hard to understand what do you mean here
you may also create a separate function for that task and it will be okay to do search each time for 3 elements